### PR TITLE
Document usage with Next.js

### PR DESCRIPTION
**Pull request by oskarhane**
# Description

This adds a guide on how to use this library in a Next.js fullstack application.  
It also adds tests for the code in the guide so it doesn't become outdated.

# Checklist

- [X] Documentation has been updated

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-graphql-docs-502.surge.sh

**Comment by oskarhane** [comment]
There's an issue with the file includes in the docs build system, I've reached out to the docs team to figure out why - it works fine locally.

**Comment by oskarhane** [comment]
The includes in the docs now work as expected.

File: docs/modules/ROOT/pages/guides/frameworks/nextjs.adoc [hunk 1]
```diff
@@ -0,0 +1,54 @@
+[[usage-nextjs]]
+
+:sourcedir: example$code-partials/nextjs
+
+= Usage with Next.js
+
+This page describes the basics on how to use `@neo4j/graphql` as part of a https://nextjs.org[Next.js] fullstack application.
```
Comment (angrykoala) [inline docs/modules/ROOT/pages/guides/frameworks/nextjs.adoc:7]: Maybe fullstack is a bit redundant if already talking about next.js :thinking: ?
```suggestion
This page describes the basics on how to use `@neo4j/graphql` as part of a https://nextjs.org[Next.js] application.  
```

**Review by angrykoala — APPROVED** [review]

File: docs/modules/ROOT/pages/guides/frameworks/nextjs.adoc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+[source, bash, indent=0]
+----
+include::{sourcedir}/add-dependencies.sh[]
+----
+
+=== Configure secrets (Neo4j credentials)
+Next.js supports `.env` files for storing secrets secrurely, so create a `.env.local` file with the following contents to set it up.
+
+Make sure you have a Neo4j DBMS running on the configured URI below. Download https://neo4j.com/download[Neo4j Desktop] for easy management of local Neo4j DBMS:s.
```
Comment (darrellwarde) [inline docs/modules/ROOT/pages/guides/frameworks/nextjs.adoc:37]: I'm guessing "DBMS:s" is a typo - I hate pluralising acronyms so an alternative wording suggestion below!

```suggestion
Make sure you have a Neo4j DBMS running on the configured URI below. Download https://neo4j.com/download[Neo4j Desktop] for easy local Neo4j DBMS management.
```

**Review by darrellwarde — CHANGES_REQUESTED** [review]
Looks good! Just a small typo which needs correcting.

**Review by darrellwarde — APPROVED** [review]
LGTM, love that we're adding these kind of guides in!

**Pull request merged**